### PR TITLE
Changelog and documentation for the improved rule module.

### PR DESCRIPTION
**Pull request by lgetwan**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Changelog for the new features of the rule module was missing.

## What is the new behavior?

- Added a changelog
- Added @geoff77 as an author
- Described the new behavior regarding idempotency in the module docs' notes.

### CODE REVIEW

File: changelogs/fragments/bugfix_rule_module.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to
 bugfixes:
-    - Rule module - Now take the folder into account when checking for existing rules.
+    - Rule module - Now properly comparing the specified rule with the existing ones to achieve idempotency.
```
Comment (robin-tribe29) [inline changelogs/fragments/bugfix_rule_module.yml:4]: Maybe we want to use the `known_issues` keyword to hint at the "limitations" of the module in the current version?

File: plugins/modules/rule.py [hunk 2]
```diff
@@ -91,6 +89,17 @@
...
+notes:
+    - To achieve idempotency, this module is comparing the specified rule with the already existing
+      rules based on conditions, folder, value_raw and enabled/disabled.
+    - To be able to compare the value_raw, which is internally stored in python format, the module
+      has to do a workaround: it is creating the specified rule, and then compares this rule with 
+      all existing rules.
+    - Then, in case of I(state) == absent, it will delete both rules: the specified one and the duplicate
+      found.
+    - Or, in case of I(state) == present, it will delete the new rule, if a duplicate is already there,
+      and keep it if not.
```
Comment (robin-tribe29) [inline plugins/modules/rule.py:102]: I like the explanation, I think it reads rather well.
We might want to add, that as a side effect, changes are created in Checkmk.

File: changelogs/fragments/bugfix_rule_module.yml (hunk 1 above)
Comment (lgetwan) [reply changelogs/fragments/bugfix_rule_module.yml:4]: Done.

File: plugins/modules/rule.py (hunk 2 above)
Comment (lgetwan) [reply plugins/modules/rule.py]: Also done.

**Pull request merged**